src/cmd/services/m3coordinator/downsample/downsampler.go

```diff
@@ -0,0 +1,756 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```

Comment:
I'm not sure if this belongs in m3aggregator. This is essentially using m3aggregator as a library and this is the code that configures the various config options and creates the aggregator, which IMO should live outside the aggregator library. Additionally, there's a lot of logic here that parses tags etc, which should live outside m3aggregator since the aggregator has no notion of tags and it would be strange to introduce it there.

Reply:
This downsampler is essentially a single node aggregation package, currently it's not used outside of the coordinator and I doubt would really have a home anywhere else (also the only two options we should ever really offer is simple, non-HA downsampling in the m3coordinator, or proper downsampling with the m3aggregator in a clustered setup).  

I'm inclined to leave it here until it's used anywhere else (and hopefully never used anywhere else, since you should be using the m3aggregator otherwise).  Also it helps avoid needing to make it super generic and optimize it for it's use in the m3coordinator.